pkgs/by-name/kr/kraken2/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  rsync,
+  perl,
+  wget,
+  llvmPackages,
+  zlib,
+  python3,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "kraken2";
+  version = "2.1.5";
+
+  src = fetchFromGitHub {
+    owner = "DerrickWood";
+    repo = "kraken2";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-9ZW29SXWzy/fZreOmNnjywwlaSReZXnqq+uzGsuWP2g=";
+  };
+
+  buildInputs = [
+    rsync
+    perl
+    wget
+    llvmPackages.openmp
+    zlib
+    python3
+  ];
+
+  installFlags = [ "KRAKEN2_DIR=$(out)/libexec/kraken2" ];
+
+  postInstall = ''
+    mkdir -p $out/bin
+    ln -s $out/libexec/kraken2/kraken2 $out/bin
+    ln -s $out/libexec/kraken2/kraken2-build $out/bin
+    ln -s $out/libexec/kraken2/kraken2-inspect $out/bin
+    ln -s $out/libexec/kraken2/k2 $out/bin
+  '';
```

The scripts have some extra dependencies (i.e., wget) and expect the `libexec/kraken2` directory to be on the path.
It's probably most maintainable to wrap them all then to try and patch them individually, e.g.,:
```suggestion
  for file in $out/libexec/kraken2/* ; do
    if [ -x "$file"] ; then
      wrapProgram "$file" \
        --prefix PATH : $out/libexec/kraken2:${
          lib.makeBinPath [
            wget
            perl
          ] 
        }
      fi
    done
  '';
```